Change `composition checks` to `build checks` (#560)
apollographql/docs: src/content/graphos/delivery/schema-checks.mdx

> **Note:** If you've [integrated schema checks with your GitHub PRs](#integrating-with-github), a rerun of the check also updates the status of the check in Github.

## Build checks

> When you run `rover subgraph check`, GraphOS performs a **build check** _before_ it performs any other checks. Also known as a **composition check**, a build check verifies that changes you make to a subgraph schema are valid GraphQL definitions and are compatible with your _other_ subgraph schemas, enabling them to compose into a _supergraph_ schema for your router ([learn more](/federation/federated-types/composition/)).

Reviewer: I tried to make this a little more readable:

```suggestion
When you run `rover subgraph check`, GraphOS first performs a **build check** to verify that the subgraph schema still contains valid GraphQL after any changes. Build checks are also known as a **composition checks** because they verify that changes to the subgraph are compatible with your _other_ subgraph schemas, enabling them to compose into a _supergraph_ schema for your router ([learn more](/federation/federated-types/composition/)).
```

This doesn't explicitly state that build checks are performed before other checks (though "first" implies that), but I think that's alright since it's called out directly below.

> > If a build check fails, Studio does _not_ then perform additional checks for the provided schema.

Reviewer: ```suggestion
> If a build check fails, Studio does _not_ perform additional checks for the provided schema.
```
